bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/progress/ProgressManager.java

```diff
@@ -1030,6 +1029,8 @@ public Image getIconFor(Job job) {
 	private void setUserInterfaceActive(boolean active) {
 		IWorkbench workbench = PlatformUI.getWorkbench();
 		Shell[] shells = workbench.getDisplay().getShells();
+		while (workbench.getDisplay().readAndDispatch())
```

Comment:
Event pumper not always guarantee that all events processed. Some events can be processed after the loop.
It will work in most cases. But not the rule of thumb.